code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -628,34 +656,94 @@ void handleEdgeResponse(@NonNull final Event event) {
+                                + " event is null or event data is null/ empty.");
+                return;
+            }
+
+            final String errorType =
+                    DataReader.optString(
+                            eventData,
+                            OptimizeConstants.Edge.ErrorKeys.TYPE,
+                            OptimizeConstants.ERROR_UNKNOWN);
+            final int errorStatus =
+                    DataReader.optInt(
+                            eventData,
+                            OptimizeConstants.Edge.ErrorKeys.STATUS,
+                            OptimizeConstants.UNKNOWN_STATUS);
+            final String errorTitle =
+                    DataReader.optString(
+                            eventData,
+                            OptimizeConstants.Edge.ErrorKeys.TITLE,
+                            OptimizeConstants.ERROR_UNKNOWN);
+            final String errorDetail =
+                    DataReader.optString(
+                            eventData,
+                            OptimizeConstants.Edge.ErrorKeys.DETAIL,
+                            OptimizeConstants.ERROR_UNKNOWN);
+            final Map<String, Object> errorReport =
+                    DataReader.optTypedMap(
+                            Object.class,
+                            eventData,
+                            OptimizeConstants.Edge.ErrorKeys.REPORT,
+                            new HashMap<>());
+
+            Log.warning(
                     OptimizeConstants.LOG_TAG,
                     SELF_TAG,
-                    "handleEdgeErrorResponse - Ignoring the Edge error response event, either event"
-                            + " is null or event data is null/ empty.");
-            return;
+                    "handleEdgeErrorResponse - Decisioning Service error! Error type: (%s),\n"
+                            + "title: (%s),\n"
+                            + "detail: (%s),\n"
+                            + "status: (%s),\n"
+                            + "report: (%s)",
+                    errorType,
+                    errorTitle,
+                    errorDetail,
+                    errorStatus,
+                    errorReport);
+
+            // Check if the errorStatus is in the list of recoverable error codes
+            if (recoverableNetworkErrorCodes.contains(errorStatus)) {
+                Log.debug(
+                        OptimizeConstants.LOG_TAG,
+                        SELF_TAG,
+                        "Recoverable error encountered: Status %d",
+                        errorStatus);
+                return;
+            } else {
+                AEPOptimizeError aepOptimizeError =
+                        new AEPOptimizeError(
+                                errorType, errorStatus, errorTitle, errorDetail, errorReport, null);
+                updateRequestEventIdsErrors.put(requestEventId, aepOptimizeError);
+            }
+        } catch (final Exception e) {
```

Comment:
nit: We're using DataReader.opt which does not throw DataReader exception. Can you verify if we need a try-catch here?